### PR TITLE
Don't wait for CTRL-D before starting up server

### PR DESCRIPTION
**Pull request by runarorama**
Fixes an issue where setting the `UCM_PORT` in the environment would cause you to have to send `EOF` to ucm in order to allow the server startup to complete.

### CODE REVIEW

**Review by pchiusano — APPROVED** [review]
LGTM though I don't understand from the diff why the old code behaved as it did and why the new code does the right thing. 🙃

File: parser-typechecker/src/Unison/Server/CodebaseServer.hs [hunk 1]
```diff
@@ -152,23 +152,18 @@ start
...
+  let settings = appEndo
+        (  foldMap (Endo . setPort)              envPort
+        <> foldMap (Endo . setHost . fromString) envHost
+        )
+        defaultSettings
       a = app codebase token
-
-  case envPort of
-    Just p -> do
-      (k token p)
```
Comment (runarorama) [inline parser-typechecker/src/Unison/Server/CodebaseServer.hs:168]: This starts up UCM and waits for it to finish (which it appears to do when it receives a CTRL-D for some reason.

File: parser-typechecker/src/Unison/Server/CodebaseServer.hs [hunk 2]
```diff
@@ -152,23 +152,18 @@ start
...
-      runSettings (setPort p settings) a
```
Comment (runarorama) [inline parser-typechecker/src/Unison/Server/CodebaseServer.hs:169]: This starts the server after the UCM CLI has "finished".

File: parser-typechecker/src/Unison/Server/CodebaseServer.hs [hunk 3]
```diff
@@ -152,23 +152,18 @@ start
...
-    Nothing ->
-      withApplicationSettings settings (pure a) (k token)
+  withApplicationSettings settings (pure a) (k token)
```
Comment (runarorama) [inline parser-typechecker/src/Unison/Server/CodebaseServer.hs:166]: Start the server and once it has started, fire up the CLI.

**Pull request merged**